### ISSUE TITLE
[LAP Rescue Run] Execute preregistered LS0.05 seed42 validation-only rescue

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#60]
## Objective

Execute exactly one **LAP-GNN seed-42 validation-only rescue experiment** using the reviewed label-smoothing adapter merged from PR #59.

Scientific question:

> Does training-only label smoothing with epsilon `0.05` materially reduce the frozen LAP-GNN train-validation macro-F1 gap while preserving or improving the existing validation accuracy/macro-F1 under the same graph/model/optimizer/scheduler/checkpoint protocol?

This issue authorizes **one scientific train/validation attempt only** after all preflight gates pass. It does not authorize test access, epsilon sweep, architecture changes, DropMessage/DropEdge, optimizer/scheduler changes, resume, or outcome-driven retry.

## Exact execution source

Run only from exact merged `main` commit:

`6d89d17b2d3c39b7bf57084f9de4b707a92eb73e`

Reviewed PR #59 head:

`58c4ce1781928ac7d2058d0e59c9ca63ecfb7c4e`

Candidate package:

`research/candidates/tf_lap_gnn_ls005_rescue/`

Registered candidate identities:

- config SHA-256: `bb89c1df7d563ec26a74469c86091fc58cb9f60698d447f24ea2589cdd605535`
- loss adapter SHA-256: `72435f59cbc138f4e2184b384d0a4e2c85cdd0082655a6371e0a61ee6362e1e5`
- validation wrapper SHA-256: `8afb489e1fd5444f50c4be48d25e12157ecb6759e7f29521eeb17d9e35697216`
- implementation evidence SHA-256: `0a3fdc27fa14375c660f22cf4ab9f73373c9750a6103acb34c74ecc2d1f14455`

Frozen package remains read-only:

`standalone/lap_gnn_tensorflow_ofix7_mid_candidate/`

Frozen locks:

- scientific payload SHA-256: `286be711a53b76511bcf3b9bf949fad694f7c7d272392f9defc56f4914822c0e`
- trainer SHA-256: `4c3cb1aa311578038ff656cb7d119103ae5a651135f8ee1c76e37c2c04c1fc75`
- frozen validation-only wrapper SHA-256: `c94c122066fdd19210c8ba64a2a61567b249fad4f69c69cb4236b68cce6ff7b4`
- execution contract SHA-256: `14acc2750875a25922007459161a137158d8040805e616166be923f63658bf22`
- parameters: `1,061,192`
- trainable variables: `127`
- dataset signature: `fer2013_train28709_val3589_test3589`

Any source/payload/model identity mismatch => STOP before training.

## Single scientific intervention

The only substantive change relative to the frozen seed42 baseline is the **training objective target**:

`epsilon = 0.05`

For each integer label `y`, seven classes:

`target = (1 - 0.05) * one_hot(y, 7) + 0.05 / 7`

Training loss is float32 softmax cross entropy with these targets.

Critical scope boundary:

- only `lap_gnn_tf.training.execution.sparse_cross_entropy` is temporarily rebound for the registered training-step construction/execution window;
- validation and clean-train evaluation remain frozen hard CE;
- original execution binding must be restored after completion/failure;
- no frozen source file is modified.

Therefore validation loss, scheduler monitor, early stopping monitor, and clean-train evaluation loss retain frozen hard-CE semantics.

## Frozen training protocol

Except for training CE target smoothing, inherit the exact OFIX7-mid seed42 policy:

- seed `42`
- train `28,709`
- validation `3,589`
- test forbidden
- train batch `16`
- eval batch `32`
- graph workers `2`
- tf-data prefetch `2`
- tf-data parallel calls `1`
- graph cache `64`
- mixed precision ON
- XLA OFF
- memory growth ON
- Torch-compatible AdamW
- LR `3e-4`
- weight decay `1e-3`
- no gradient clipping
- max epochs `90`
- ReduceLROnPlateau on hard-CE `val_loss`, factor `0.5`, patience `5`, threshold `1e-4`, min LR `3e-5`
- early stopping on hard-CE `val_loss`, minimum epoch `30`, patience `15`
- checkpoint/model selection = earliest strict maximum `val_accuracy`
- train-only prior-corruption schedule unchanged `0.1 -> 0.2 -> 0.3`
- clean full-train evaluation every epoch
- no resume

No graph/node/edge/prior/anchor/context/motif/readout/model change.

## Registered Kaggle runtime

Require the baseline-compatible runtime class:

- Kaggle Linux
- Python `3.12.12`
- TensorFlow `2.18.1`
- Keras `3.15.0`
- CUDA `12.5.1`
- cuDNN major `9`
- exactly `2 x Tesla T4`
- no operation-determinism change

If the runtime class differs, STOP before the training command. Do not silently substitute P100/CPU or newer TensorFlow/Keras.

## Registered inputs

FER split root:

`/kaggle/input/datasets/doduyquynii/fer13-split/fer13-split`

The training invocation uses only:

`/kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv`

Validation sample identity is obtained only through the frozen validation prior/cache path. `test.csv` must never be opened/read/hashed/loaded.

MediaPipe priors:

`/kaggle/input/datasets/irthn1311/d16-mediapipe-pixel-priors-best-retry-rescue/outputs/d16_mediapipe_pixel_priors_best_retry_rescue`

Clean graph cache:

`/kaggle/input/datasets/irthn1311/ofix7-mid-seed42-records`

Only train/validation records plus shared aggregate metadata required by the frozen loader are allowed. No test prior/cache sample record may be accessed.

## Preflight

Before the command starts:

1. `git fetch origin`
2. checkout detached exact `6d89d17b2d3c39b7bf57084f9de4b707a92eb73e`
3. require exact `git rev-parse HEAD`
4. require clean tracked worktree
5. run frozen checksum verification
6. verify all candidate/frozen SHA-256 locks above
7. verify loaded candidate config deep-diff vs frozen seed42 is exactly:
   - `run_name`
   - `loss.label_smoothing: 0.0 -> 0.05`
8. verify model params/trainable-variable count `1,061,192 / 127`
9. verify training binding is frozen hard CE before adapter entry and evaluator hard CE is original
10. verify exact runtime class
11. verify registered train/prior/cache input roots without opening/hash-reading `test.csv` or test sample records
12. require a fresh absent/empty output root

Any failure => STOP BEFORE TRAINING. A preflight-only stop does not consume the scientific attempt.

## Exact authorized command

From repository root at exact detached source:

```bash
python -m research.candidates.tf_lap_gnn_ls005_rescue.train_validation_only \
  --config research/candidates/tf_lap_gnn_ls005_rescue/configs/fer2013_ofix7_mid_seed42_ls005.yaml \
  --fer-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv \
  --prior-root /kaggle/input/datasets/irthn1311/d16-mediapipe-pixel-priors-best-retry-rescue/outputs/d16_mediapipe_pixel_priors_best_retry_rescue \
  --clean-graph-cache-dir /kaggle/input/datasets/irthn1311/ofix7-mid-seed42-records \
  --output-root /kaggle/working/tf_lap_gnn_ls005_rescue_seed42 \
  --device gpu \
  --graph-workers 2 \
  --tf-data-prefetch 2 \
  --tf-data-parallel-calls 1 \
  --graph-cache-size 64 \
  --batch-size 16 \
  --eval-batch-size 32 \
  --mixed-precision \
  --no-xla \
  --memory-growth \
  --no-resume
```

No `--limit-*` arguments.

Once this exact training command starts, the one registered scientific attempt is consumed.

## One-attempt / rerun rule

After command launch, stop after normal completion, early stopping, exception, OOM, CUDA failure, Kaggle hard censor, or any other termination.

Do not automatically rerun or resume.

Any second attempt requires a new explicit research-lead amendment justified only by reviewed technical invalidity/censoring, never by disappointing metrics.

No epsilon `0.10` follow-up is automatically authorized.

## Frozen comparator

Primary same-policy baseline selected checkpoint (earliest strict max `val_accuracy`, epoch 31):

- val accuracy `0.6319308999721371`
- val macro-F1 `0.5938407974340496`
- clean-train macro-F1 `0.8207655611897143`
- macro gap `22.69247637556647 pp`

Secondary baseline best-macro trajectory reference:

- best val macro-F1 `0.601166548701511` at epoch 26
- clean-train macro-F1 at epoch 26 `0.7562805286580438`
- macro gap `15.511397995653287 pp`

No test metric is a comparator.

## Primary selected-checkpoint endpoint

Use the candidate checkpoint selected by the frozen policy: **earliest strict max validation accuracy**.

At that selected checkpoint report:

- validation accuracy
- validation macro-F1
- hard-CE validation loss
- clean-train accuracy
- clean-train macro-F1
- hard-CE clean-train loss
- selected-checkpoint accuracy gap
- selected-checkpoint macro-F1 gap
- epoch

Compute:

- `delta_val_accuracy_pp = 100 * (candidate_val_acc - 0.6319308999721371)`
- `delta_val_macro_pp = 100 * (candidate_val_macro - 0.5938407974340496)`
- `macro_gap_improvement_pp = 22.69247637556647 - candidate_macro_gap_pp`

Secondary trajectory-only diagnostics may report the best validation macro-F1 epoch/value and its clean-train macro gap, but must not replace the selected-checkpoint decision endpoint.

## Preregistered decision precedence

Use exactly the implementation-reviewed classifier:

1. `LAP_LS005_STRETCH_RESCUE`
2. `LAP_LS005_RESCUE_PASS`
3. `LAP_LS005_GENERALIZATION_RESCUE_WITH_SMALL_ACCURACY_COST`
4. `LAP_LS005_OVERREGULARIZED`
5. `LAP_LS005_NO_CLEAR_RESCUE`
6. `LAP_LS005_REGRESSION`
7. `LAP_LS005_INCONCLUSIVE`

### STRETCH_RESCUE
All at selected checkpoint:
- val accuracy `>= 0.6500`
- val macro-F1 `>= 0.6200`
- macro gap `<= 8.0 pp`

### RESCUE_PASS
All:
- val accuracy `>= 0.6319308999721371`
- val macro-F1 `>= 0.5938407974340496`
- macro gap `<= 10.0 pp`

### GENERALIZATION_RESCUE_WITH_SMALL_ACCURACY_COST
Unless higher priority applies, all:
- val accuracy `>= 0.6219308999721371`
- macro-gap improvement `>= 7.5 pp`
- val macro-F1 `>= 0.5838407974340496`

### OVERREGULARIZED
Unless higher priority applies:
- macro-gap improvement `>= 7.5 pp`
- AND val accuracy change `< -1.0 pp` OR val macro-F1 change `< -1.0 pp`

### NO_CLEAR_RESCUE
Unless higher priority applies:
- absolute val-accuracy change `< 1.0 pp`
- macro-gap improvement `< 7.5 pp`
- absolute val-macro change `<= 1.0 pp`

### REGRESSION
Unless higher priority applies:
- no qualifying gap rescue
- AND val accuracy `<= baseline - 1.0 pp` OR val macro-F1 `<= baseline - 1.0 pp`

Else `INCONCLUSIVE`.

These are practical single-seed diagnostics, not significance/equivalence claims.

## Scientific validity gates after execution

A decision is valid only if all are true:

- exact checkout/source locks PASS
- subprocess exits successfully
- frozen `VALIDATION_ONLY_COMPLETE.json` valid
- candidate `LS005_VALIDATION_ONLY_COMPLETE.json` valid
- `training_validation_completed=true`
- `final_test_skipped=true`
- `test_accessed=false`
- `test_data_constructed=false`
- `test_checkpoint_loaded=false`
- training execution binding restored
- evaluator hard-CE binding remained unchanged
- model identity `1,061,192 / 127`
- selected checkpoint and metadata exist and agree with earliest strict global max val accuracy in complete history
- train/validation history internally consistent
- no forbidden test artifacts/sample access

If any gate fails: `scientific_result_valid=false`, `scientific_interpretation=null`; do not classify performance.

## Evidence preservation

Capture merged stdout/stderr for the exact subprocess.

Preserve the complete fresh output root plus:

- pre-run manifest with checkout/source/runtime/resource identities
- exact command + return code
- subprocess log
- frozen validation-only marker
- LS005 completion marker
- `history.json`
- `train_log.csv`
- `resolved_config.json/yaml`
- `provenance.json`
- `telemetry.json`
- `latest_epoch_summary.json` / runtime progress if present
- best-val-accuracy checkpoint + weights + metadata
- selected-checkpoint recomputation/decision report

After normal completion or technical termination, package available evidence exactly once as:

`/kaggle/working/tf_lap_gnn_ls005_rescue_seed42_evidence.zip`

Report:

- ZIP SHA-256
- bytes
- member count
- outer ZIP integrity
- embedded `.keras` SHA-256
- embedded `.keras` integrity/member count
- forbidden test/CSV member scan

Do not repack after the final ZIP hash.

If Kaggle hard-censors before the archive can be published, report technical censoring; do not infer a scientific label from partial history and do not automatically resume.

## Required final report

Return:

1. exact checkout SHA and clean worktree before/after
2. candidate/frozen source SHA-256 locks
3. model params/trainable-variable count
4. runtime identity
5. exact command
6. return code / completion reason
7. epochs completed
8. selected checkpoint epoch and policy proof
9. selected checkpoint train/val accuracy, macro-F1, hard-CE loss
10. selected checkpoint accuracy/macro gaps
11. deltas vs frozen selected-checkpoint baseline
12. macro-gap improvement vs baseline
13. best validation macro trajectory epoch/value and clean-train macro gap there
14. LR reductions / early-stop state
15. registered decision label from the frozen classifier
16. completion markers / binding-restoration evidence
17. `test_access=false`
18. evidence ZIP SHA/bytes/member count/integrity
19. embedded checkpoint SHA/integrity/member count
20. warnings/errors/runtime notes

Final valid status:

`LAP_GNN_LS005_RESCUE_SEED42_COMPLETE_PENDING_INDEPENDENT_REVIEW`

Invalid/technical status:

`LAP_GNN_LS005_RESCUE_SEED42_INVALID_PENDING_INDEPENDENT_REVIEW`

## Interpretation boundary

A positive rescue label supports only that mild target smoothing improved the **fit/generalization trade-off under this frozen protocol**. It does not prove FER2013 label noise caused the original overfit gap.

A no-clear or regression result does not prove label smoothing is universally ineffective; it only rejects this preregistered epsilon/protocol as a sufficient rescue for the current submission cycle.

No outcome in this issue authorizes test access.
